//Define the used version of proto
syntax = "proto3";

package aqcalibration;

//This represents one row in the dataset. For now assume we now beforehand which fields are gonna be present in the dataset.
message DataRow {
  string time = 1;
  float64 elgeseter_pm25_nilu = 2;
  float64 torget_pm25_nilu = 3;
  float64 elgeseter_pm10_nilu= 4;
  float64 torget_pm10_nilu = 5;
  float64 elgeseter_pm1_iot = 6;
  float64 torget_pm1_iot = 7;
  float64 elgeseter_pm25_iot = 8;
  float64 torget_pm25_iot = 9;
  float64 elgeseter_pm10_iot = 10;
  float64 torget_pm10_iot = 11;
  float64 elgeseter_opctemp_iot = 12;
  float64 torget_opctemp_iot = 13;
  float64 elgeseter_opchum_iot = 14;
  float64 torget_opchum_iot = 15;
  float64 air_temperature = 16;
  float64 relative_humidity = 17;
  float64 precipitation = 18;
  float64 air_pressure = 19;
  float64 wind_speed = 20;
  float64 wind_direction = 21;
  float64 16219V72812_Total = 22;
  float64 44656V72812_Total = 23;
  float64 10236V72161_Total = 24;
}

message InputData {
  repeated DataRow rows = 1; 
}

// Request message must include: 1) the dataset; 2) the name of the sensor to use as target; 3) the pollutant (PM2.5 or PM10)
message CalibTrainRequest {
  InputData data = 1;
  string train_sensor = 2;
  string pollutant = 3;
}

message CalibTrainResponse {
  float64 rmse = 1;
}

message CalibPredictRequest {
  DataRow data = 1;
}

message CalibPredictResponse {
  float64 calibrated_val = 1;
}

//Define the service
service Calibration {
  rpc calibration_train(CalibTrainRequest) returns (CalibTrainResponse);
  rpc calibration_predict(CalibPredictRequest) returns (CalibPredictResponse);
}